packages/core/utils/lib/errors.js

```diff
@@ -55,19 +55,28 @@ class ForbiddenError extends ApplicationError {
   }
 }
 
-class PayloadTooLargeError extends ApplicationError {
+class UnauthorizedError extends ApplicationError {
   constructor(message, details) {
     super(message, details);
-    this.name = 'PayloadTooLargeError';
-    this.message = message || 'Entity too large';
+    this.name = 'UnauthorizedError';
+    this.message = message || 'Unauthorized';
   }
 }
 
-class UnauthorizedError extends ApplicationError {
+class RateLimitError extends ApplicationError {
   constructor(message, details) {
     super(message, details);
-    this.name = 'UnauthorizedError';
-    this.message = message || 'Unauthorized';
+    this.name = 'RateLimitError';
+    this.message = message || 'Too many requests, please try again later.';
+    this.status = 429;
```

Comment:
To put the 429 status error you need to edit this file instead : `packages/core/strapi/lib/services/errors.js`